# Constructs the 'stats' data structure, which is a data frame with a row for
# each COG in genes_coverage.  Each COG has data already calculated for it.
# Information is added to this data structure for subsequent analysis.

library(mclust)

maps_to_pert <- function(row){
    as.numeric(substr(as.character(row[[1]]), start=11, stop=13))
}

genes <- readLines("~/Code/research/genes_coverage")
stats <- data.frame( array(0, dim=c(length(genes), 13)))
colnames(stats) <- c('gene', 'min_ari', 'max_ari', 'ave_ari', 'moved:total reads', 'average_moved' )
stats[,1] <- genes
for (g in 1:(length(genes))){
    # Add gene name.
    gene <- genes[g]
    # Retrieve the "statfile" created by perturb.py, which tracked the number
    # of reads per sample, the number moved in that sample, and the random 
    # seed fed to perturb.py to produce those movements ... for a single gene.
    sf <- read.csv(paste("~/outfiles/", gene, "/statfile", sep=""), header=F, skip=1, col.names=c('sample','moved','total','seed'))
    # Calculate the ratio of moved:total reads.
    stats[g,5] <- sum(sf[,2])/sum(sf[,3])
    # Calculate the average number of reads moved for each gene across all 
    # perturbations.
    moved <- array(0, dim=10)
    for (rowi in 1:(nrow(sf))){
        moved[maps_to_pert(sf[rowi,])] <- moved[maps_to_pert(sf[rowi,])] + sf[rowi,2]
    }
    stats[g,6] <- mean(moved)
    
    ### Stability analysis
    # Inspect the cut3_pert0 file for each gene, generated by perturb.py, which
    # is the squash clustering of the unperturbed data set, cut into 3 clusters.
    fc0 <- paste("~/outfiles/", gene, "/cut3_pert0", sep="")
    c0 <- read.csv(fc0, header=T, sep=" ")
    pert <- rep(0,10)
    # For the same gene, repeat for the other 10 perturbations.  The perturbed
    # clusterings are each compared with the unperturbed clustering c0 using the
    # adjusted Rand Index, yielding 10 measures of similarity.
    for (i in 1:10){
        fci <- paste("~/outfiles/", gene, "/cut3_pert", as.character(i), sep="")
        ci <- read.csv(fci, header=T, sep=" ")
        pert[i] <- adjustedRandIndex(as.numeric(c0[1,]), as.numeric(ci[1,]))
    }    
    # Calculate the min, the max, and the mean of the 10 measures for this gene.
    # The mean is the most important for subsequent analysis.
    stats[g,2] <- min(pert)
    stats[g,3] <- max(pert)
    stats[g,4] <- mean(pert)
}

# Append to stats several other columns from various csv files
# * total_edpls.csv: total_reads, total_EDPL, and total branchlength of tree.
#   total_edpls.csv is produced by compute_total_edpl.sh.
# * gut_pairwise_identities.csv is the percent pairwise identity for each gene's
#   multiple sequence alignment, as calculated by geneious 7.0.4
# * all_ave_phylo_diversity.csv is the average phylogenetic diversity of across
#   a gene's samples.  Produced by calc_fpd.sh
stats[,7:9] <- read.csv("~/outfiles/total_edpls.csv", header=T)[,2:4]
stats[,10] <- read.csv("gut_pairwise_identities.csv", header=T)[,2]
stats[,11] <- read.csv("~/outfiles/all_ave_phylo_diversity.csv", header=T)[,2]
stats[,12:13] <- read.csv("gene_convexities.csv", header=T)[,2:3]
 
colnames(stats)[7:13] <- c("total_reads","total_edpl","branch_length","pairwise_percent_identity","ave_phylo_div", "Convex Phylum-Taxa in Tree", "Total Taxa in Tree")
